examples/tv-casting-app/android/App/app/src/main/java/com/matter/casting/DiscoveryExampleFragment.java

```diff
@@ -0,0 +1,390 @@
+
+    // Remove the CastingPlayerChangeListener
+    Log.i(TAG, "stopDiscovery() removing CastingPlayerChangeListener");
+    MatterError errRemove =
+        matterCastingPlayerDiscovery.removeCastingPlayerChangeListener(castingPlayerChangeListener);
+    if (errRemove.hasError()) {
+      Log.e(
+          TAG,
+          "stopDiscovery() matterCastingPlayerDiscovery.removeCastingPlayerChangeListener() called, errRemove: "
+              + errRemove);
+      stopDiscoverySuccess = false;
+    }
+
+    if (!stopDiscoverySuccess) {
+      Log.e(TAG, "stopDiscovery() Warning: complete with errors! Discovering: " + discovering);
+    }
+  }
+}
+
+class CastingPlayerCommissionerAdapter extends ArrayAdapter<CastingPlayer> {
+  private final List<CastingPlayer> playerList;
+  private final Context context;
+  private LayoutInflater inflater;
+  private static final String TAG = CastingPlayerCommissionerAdapter.class.getSimpleName();
+
+  public CastingPlayerCommissionerAdapter(Context context, List<CastingPlayer> playerList) {
+    super(context, 0, playerList);
+    Log.i(TAG, "CastingPlayerCommissionerAdapter() constructor called");
+    this.context = context;
+    this.playerList = playerList;
+    inflater = (LayoutInflater.from(context));
+  }
+
+  @Override
+  public View getView(int i, View view, ViewGroup viewGroup) {
+    view = inflater.inflate(R.layout.commissionable_player_list_item, null);
+    String buttonText = getCastingPlayerButtonText(playerList.get(i));
+    Button playerDescription = view.findViewById(R.id.commissionable_player_description);
+    playerDescription.setText(buttonText);
+
+    View.OnClickListener clickListener =
+        v -> {
+          CastingPlayer castingPlayer = playerList.get(i);
+          Log.d(
+              TAG,
+              "OnItemClickListener.onClick() called for castingPlayer with deviceId: "
+                  + castingPlayer.getDeviceId());
+          DiscoveryExampleFragment.Callback callback1 = (DiscoveryExampleFragment.Callback) context;
+          // TODO: In following PRs. Implement CastingPlayer connection
+          // callback1.handleCommissioningButtonClicked(castingPlayer);
+        };
+    playerDescription.setOnClickListener(clickListener);
+    return view;
+  }
+
+  private String getCastingPlayerButtonText(CastingPlayer player) {
+    String main = player.getDeviceName() != null ? player.getDeviceName() : "";
+    String aux = "" + (player.getDeviceId() != null ? "Device ID: " + player.getDeviceId() : "");
+    aux +=
+        player.getProductId() > 0
+            ? (aux.isEmpty() ? "" : ", ") + "Product ID: " + player.getProductId()
+            : "";
+    aux +=
+        player.getVendorId() > 0
+            ? (aux.isEmpty() ? "" : ", ") + "Vendor ID: " + player.getVendorId()
+            : "";
+    aux +=
+        player.getDeviceType() > 0
+            ? (aux.isEmpty() ? "" : ", ") + "Device Type: " + player.getDeviceType()
+            : "";
```

Comment:
After device type, can you add a boolean \`\`\`resolvedIP\`\`\` - set it to true if the CastingPlayer has at least one IP Address, false otherwise.